src/platform/packages/private/kbn-esql-editor/src/custom_commands/create_lookup_index.tsx

```diff
@@ -11,46 +11,123 @@ import type { AggregateQuery } from '@kbn/es-query';
 import { EditLookupIndexContentContext } from '@kbn/index-editor';
 import { useKibana } from '@kbn/kibana-react-plugin/public';
 import { monaco } from '@kbn/monaco';
-import React, { useCallback, useEffect, useMemo, useRef } from 'react';
+import React, { useCallback, useEffect, useRef } from 'react';
 import { css } from '@emotion/react';
 import { useEuiTheme } from '@elastic/eui';
-import { type ESQLSource } from '@kbn/esql-ast';
-import { Parser } from '@kbn/esql-ast/src/parser/parser';
-import { memoize } from 'lodash';
+import { mutate, type ESQLSource, BasicPrettyPrinter } from '@kbn/esql-ast';
+import { Parser } from '@kbn/esql-ast';
 import { IndexAutocompleteItem } from '@kbn/esql-types';
+import { i18n } from '@kbn/i18n';
 import type { ESQLEditorDeps } from '../types';
 
 /**
- * Returns a query with appended index name to the join command.
+ * Replace the index name in a join command.
  *
- * @param query Input query
- * @param cursorPosition
- * @param indexName
+ * @param query             - full ES|QL query
+ * @param initialIndexOrPos - either the index name we want to replace OR a Monaco
+ *                            cursor Position pointing at that name
+ * @param createdIndexName – new lookup-index name
  *
  * @returns {string} Query with appended index name to the join command
  */
 export function appendIndexToJoinCommand(
   query: string,
-  cursorPosition: monaco.Position,
-  indexName: string
+  initialIndexOrPos: string | monaco.Position,
+  createdIndexName: string
 ): string {
-  const cursorColumn = cursorPosition?.column ?? 1;
-  const cursorLine = cursorPosition?.lineNumber ?? 1;
+  if (!createdIndexName) return query; // nothing to do
+
+  // Resolve the “target” index name
+  let targetName: string | undefined;
+
+  if (typeof initialIndexOrPos === 'string') {
+    targetName = initialIndexOrPos.trim();
+  }
+
+  // If we came through name-path, and it equals new name – nothing to do
+  if (typeof initialIndexOrPos === 'string' && targetName === createdIndexName) {
+    return query;
+  }
+
+  // Parse and walk the AST
+  const { root } = Parser.parse(query);
+
+  // Compute cursor offset once (if needed)
+  const cursorOffset: number | undefined =
+    typeof initialIndexOrPos === 'object'
+      ? (() => {
+          const { lineNumber, column } = initialIndexOrPos;
+          const lines = query.split('\n');
+          let off = 0;
+          for (let i = 0; i < lineNumber - 1; i++) {
+            off += lines[i].length + 1;
+          }
+          return off + column - 1;
+        })()
+      : undefined;
+
+  // Pick the JOIN command to modify
+  let selectedJoin: any | undefined;
```

Comment:
If you have time to go throw the type errors: 

```suggestion
  let selectedJoin: {
    joinCmd: ESQLAstJoinCommand;
    src?: ESQLSource;
    firstArg?: ESQLSingleAstItem;
  } | undefined;
```

Reply:
update the types is https://github.com/elastic/kibana/pull/229320/commits/31c70f0bc53a2e16e8a6252c8fa0703895f7193d